src/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs

```diff
@@ -338,15 +356,34 @@ public int Indentation
             {
                 if (value < 0)
                     throw new ArgumentException(SR.Xml_InvalidIndentation);
-                _indentation = value;
+                if(_indentation != value)
+                {
+                    _indentation = value;
+                    int arraySize = _indentation * IndentArrayMultiplier;
+                    if (arraySize > _indentChars.Length)
+                    {
+                        char indentChar = _indentChars[0];
+                        _indentChars = new char[arraySize];
+                        for (int i = 0; i < arraySize; i++)
+                        {
+                            _indentChars[i] = indentChar;
+                        }
+                    }
+                }
             }
         }
 
         // Gets or sets which character to use for indenting when Formatting is set to "Indented".
         public char IndentChar
         {
-            get { return _indentChar; }
-            set { _indentChar = value; }
+            get { return _indentChars[0]; }
+            set
+            {
+                for (int i = 0; i < _indentChars.Length; i++)
+                {
+                    _indentChars[i] = value;
```

Comment:
Do we want to copy `s_defaultIndentChars` to `_indentChars` instead of do assignment in the constructor?